classes/file_validator.php

```diff
@@ -108,6 +121,13 @@ public function __construct(array $userids = [], role_assignments $assignments =
         $this->assignments    = $assignments ?: new role_assignments();;
     }
 
+    /**
+     * @return array
+     */
+    public static function whitelist() {
```

Comment:
I tried to do this

const WHITELIST = self::BOTH_WHITELIST + self::TEACHER_WHITELIST but for some reason it didn't include everything.